bootstore/src/schemes/v0/share_pkg.rs

```diff
@@ -95,31 +83,50 @@ pub struct SharePkgV0 {
     pub encrypted_shares: Vec<u8>,
 }
 
-/// An analog to [`SharePkgV0`] for nodes that were added after rack
-/// initialization. There is no encrypted_shares or nonces because of this.
-#[derive(Clone, PartialEq, Serialize, Deserialize, Zeroize, ZeroizeOnDrop)]
-pub struct LearnedSharePkgV0 {
+impl SharePkg {
+    pub fn decrypt_shares(
+        &self,
+        rack_secret: &RackSecret,
+    ) -> Result<Secret<Vec<Vec<u8>>>, TrustQuorumError> {
+        let cipher =
+            derive_encryption_key(&self.rack_uuid, &rack_secret, &self.salt);
+        decrypt_shares(self.nonce, &cipher, &self.encrypted_shares)
+    }
+}
+
+/// An analog to [`SharePkg`] for nodes that were added after rack
+/// initialization. There are no encrypted_shares or nonces because of this.
+#[derive(
+    Clone, PartialEq, Eq, Serialize, Deserialize, Zeroize, ZeroizeOnDrop,
+)]
+pub struct LearnedSharePkg {
     #[zeroize(skip)]
```

Comment:
Would it be worth moving the shared fields between `LearnedSharePkg` and `SharePkg` into a separate struct, contained within each of them?

Reply:
Good call! Done in [1ed56ca](https://github.com/oxidecomputer/omicron/pull/3385/commits/1ed56ca32063b183d2e7666b4b1fea23cde8d590)